recipes/libpq/all/conanfile.py

```diff
@@ -113,19 +114,33 @@ def build(self):
                 with tools.environment_append({"CONFIG": config}):
                     with tools.chdir(os.path.join(self._source_subfolder, "src", "tools", "msvc")):
                         self.run("perl build.pl libpq")
+                        if not self.options.shared:
+                            self.run("perl build.pl libpgport")
         else:
             autotools = self._configure_autotools()
             with tools.chdir(os.path.join(self._source_subfolder, "src", "backend")):
                 autotools.make(target="generated-headers")
             with tools.chdir(os.path.join(self._source_subfolder, "src", "common")):
                 autotools.make()
+            if tools.Version(self.version) >= "12":
+                with tools.chdir(os.path.join(self._source_subfolder, "src", "port")):
+                    autotools.make()
             with tools.chdir(os.path.join(self._source_subfolder, "src", "include")):
                 autotools.make()
             with tools.chdir(os.path.join(self._source_subfolder, "src", "interfaces", "libpq")):
                 autotools.make()
             with tools.chdir(os.path.join(self._source_subfolder, "src", "bin", "pg_config")):
                 autotools.make()
 
+    def _remove_unused_libraries_from_package(self, is_shared):
+        path = None
+        if is_shared:
+            path = os.path.join(self.package_folder, "lib", "*.a")
+        else:
+            path = os.path.join(self.package_folder, "lib", "libpq.so*")
+        for file in glob.glob(path):
+            os.remove(file)
```

Comment:
Does it properly clean up unused libraries with MinGW?

Reply:
Certainely not :+1:

Reply:
What file extension would need to be added?

Reply:
@syoliver would be so kind as to explain the solution? I have little to no experience with MinGW so if there's something wrong I would not be able to know.

Reply:
I would like to propose something but I can't test it right now. I am not sure but I think that naming is :
- *.a : static library
- *.dll : dynamic library
- *.dll.a : import library

If you can't fix it, I will try to propose something... But not sure when.

Reply:
MinGW is not a supported platform, so it should not block this PR =)

Reply:
@SpaceIm @madebr I don't know what to do with the following error, I did something bad ?

```
C:\Dev\conan-center-index\recipes\libpq\all>conan create . libpq/12.2@ --profile
=mingw --build
Exporting package recipe
libpq/12.2 exports: File 'conandata.yml' found. Exporting it...
libpq/12.2 exports: Copied 1 '.yml' file: conandata.yml
libpq/12.2: The stored package has not changed
libpq/12.2: Exported revision: 0c097720baf42d6d959a7dda17217f33
Configuration:
[settings]
arch=x86
arch_build=x86
build_type=Debug
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=9.2
os=Windows
os_build=Windows
[options]
[build_requires]
[env]

zlib/1.2.11: Forced build from source
libpq/12.2: Forced build from source
msys2/20190524: Forced build from source
7zip/19.00: Forced build from source
lzma_sdk/9.20: Forced build from source
make/4.2.1: Forced build from source
ERROR: Loop detected in context host: 'lzma_sdk/9.20' requires 'msys2/20190524'
which is an ancestor too
```

Reply:
@syoliver See https://github.com/conan-io/conan-center-index/issues/1422#issuecomment-637639426
Does removing the `msys2` build requirement for `lzma_sdk` fix this?

Reply:
related to https://github.com/conan-io/conan-center-index/issues/1422

use `-b missing` instead of `--build`

Reply:
Actually I think msys2 should not depend on 7zip. Here is the PR #1810

Reply:
@prince-chrismc 
Please consider https://github.com/prince-chrismc/conan-center-index/pull/4